dbux-data/src/dataProviderUtil.js

```diff
@@ -304,7 +304,11 @@ export default {
     }
 
     const valueRef = dp.util.getTraceValueRef(traceId);
-    return valueRef?.value;
+    if (!valueRef) {
+      // TODO: better distinguish between existing and non-existing values
+      return undefined;
```

Comment:
Let's just `throw new Error` here. I (accidentally) already added a comment on my branch to the function, warning users to call `doesTraceHaveValue` before using this function (which we do already).

Reply:
After some research on these `trace value related functions`, I just found that:
1. `doesTraceHaveValue` and `getTraceValue` are doing almost the same thing. 
2. `doesTraceHaveValue` seems not be able to handle trace with value `undefined`. A better fix will be check if the 'value' or 'valueId' property is in the trace.
3. `getTraceRef` does not throw error when 'valueId' not exist(return `null` instead), will that be inconsistent if we make `getTraceValue` throws error?